unittests/fork_db_tests.cpp

```diff
@@ -137,4 +140,46 @@ BOOST_AUTO_TEST_CASE(add_remove_test) try {
    BOOST_TEST(branch[1] == bsp11a);
 } FC_LOG_AND_RETHROW();
 
+
+// test `fork_database_t::validated_block_exists() const` member
+// -------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE(validated_block_exists, generate_forkdb_state) try {
+   bsp12b->set_valid(true);
+   bsp13b->set_valid(true);
+   bsp14b->set_valid(true);
+
+   BOOST_REQUIRE_EQUAL(true,  forkdb.validated_block_exists(bsp14b->id(), bsp14b->id()));
+   BOOST_REQUIRE_EQUAL(true,  forkdb.validated_block_exists(bsp14b->id(), bsp13b->id()));
+   BOOST_REQUIRE_EQUAL(true,  forkdb.validated_block_exists(bsp14b->id(), bsp12b->id()));
+   BOOST_REQUIRE_EQUAL(true,  forkdb.validated_block_exists(bsp14b->id(), bsp11b->id()));
```

Comment:
Giving `bsp11a->set_valid(true)`
`forkdb.validated_block_exists(bsp14b->id(), bsp13a->id()));`
will return `true`, I think.

Reply:
As it should. The call is in error because `claimed_id` is expected to be an ancestor of `id`.